manipulation/build.gradle.kts

```diff
@@ -90,3 +92,25 @@ val functionalTest = task<Test>("functionalTest") {
 tasks.getByName("check") {
     dependsOn(functionalTest)
 }
+
+tasks.whenTaskAdded {
+    if (name == "publishPluginJar") {
+        dependsOn("spotlessJava")
+    }
+}
+
+tasks.getByName("delombok") {
+    dependsOn("spotlessJava")
+}
+
+tasks.getByName("test") {
+    dependsOn("shadowJar")
+}
+
+tasks.getByName("functionalTest") {
+    dependsOn("shadowJar")
+}
+
+tasks.getByName("publishShadowPublicationToMavenLocal") {
+    dependsOn("publishPluginJavaDocsJar", "publishPluginJar", "jar")
+}
```

Comment:
I think I'd like to understand why this is needed as well.

Reply:
This happens when using `publishToMavenLocal`. I assume `publish` would have the same issues, but I can't exactly test a real publish. We only `publish` officially under Gradle 5.6, but I think we might `publishToMavenLocal` for any version.

Reply:
Is it possible to put all the extra dependency instructions in the top level build file in one place?

Reply:
Actually, I guess if we already have stuff in those files then that comment can be ignored :-)

Reply:
Perhaps similar to my prior comment ; a single line explaining why we needed this change?

Reply:
I wasn't sure how to reference the tasks in the main file, either, when I tried.

In that case, I will have to add a comment to three or four different files.

Reply:
+1, a small comment will make future maintenance easier to understand why things were added IMHO

Reply:
OK, I also did some slight rearranging so they're all located at the very bottom of their respective files now, with the same comment before the section.